Source/SolitareView.cpp

```diff
@@ -39,3 +89,93 @@ BMessage* NewSolitareOptionMessage(uint32 what) {
 	option_message->AddMessage(OPTION_MESSAGE_LABEL, message);
 	return option_message;
 }
+
+void SolitareView::_GenerateBoard()
+{
+	return;
+}
+
+void SolitareView::_LoadBitmaps()
+{
+	BString suits[] = {
+		"spade",
+		"heart",
+		"club",
+		"diamond"
+	};
+
+	// load images
+	BString filename;
+	for (short i = 0; i < CARDS_IN_SUIT; i++) {
+		for (short j = 0; j < 4; j++) {
+			filename = "";
+			filename << "Artwork/" << i + 1 << "_" << suits[j] << ".png";
+			fCards[j * CARDS_IN_SUIT + i]
+				= BTranslationUtils::GetBitmap('rGFX', filename);
+		}
+	}
+	fBack[0] = BTranslationUtils::GetBitmap('rGFX', "Artwork/back.png");
+	fEmpty = BTranslationUtils::GetBitmap('rGFX', "Artwork/empty.png");
+
+	// load audio
+	fResources = be_app->AppResources();
+	fShuffle = _LoadSound("Artwork/shuffle.wav");
+	fFanfare = _LoadSound("Artwork/fanfare.wav");
+	
+	// cache multiple backs in a row
+	for (short i = 1; i < CACHED_BACKS; i++) {
+		fBack[i] = new BBitmap(BRect(0, 0, CARD_WIDTH - 1,
+			CARD_HEIGHT + i * 18), fBack[0]->ColorSpace(), true);
+		
+		BView* fBackView = new BView(fBack[i]->Bounds(), NULL, 0, 0);
+		BRect destRect = fBack[0]->Bounds();
+		fBack[i]->AddChild(fBackView);
+		fBack[i]->Lock();
+		
+		fBackView->SetDrawingMode(B_OP_COPY);
+		fBackView->DrawBitmap(fBack[0], destRect);
+		destRect.top = i * 18;
+		destRect.bottom = destRect.top + CARD_HEIGHT;
+		fBackView->DrawBitmap(fBack[0], destRect);
+		
+		fBackView->SetDrawingMode(B_OP_ALPHA);
+		for (short j = 0; j < i + 1; j++) {
+			destRect.top = j * 18;
+			destRect.bottom = destRect.top + CARD_HEIGHT;
+			fBackView->DrawBitmap(fBack[0], destRect);
+		}
+		fBackView->Sync();
+		fBack[i]->Unlock();
+	}
+
+	Invalidate();
+}
+
+BSimpleGameSound* SolitareView::_LoadSound(const char* resourceName)
+{
+	size_t size;
+	const void* data = fResources->LoadResource('rSFX', resourceName, &size);
+	if (data == NULL) {
+		printf("Sound resource not found: %s\n", resourceName);
+		return new BSimpleGameSound("");
+	}
+	
+	gs_audio_format format;
+	format.frame_rate = 22050;
+	format.channel_count = 1;
+		// mono
+	format.format = 0x2;
+	format.byte_order = 0;
+	format.buffer_size = 0;
+		// auto
+	
+	BSimpleGameSound* sound = new BSimpleGameSound(data, size / 2, &format);
+	
+	status_t status = sound->InitCheck();
+	if (status != B_OK) {
+		printf("Error loading sound resource: %s. Error code: %d\n",
```

Comment:
Usually status_t error codes are converted to strings, using something like this:

```suggestion
		printf("Error loading sound resource %s: %s.\n", resourceName, strerror(status));
```

Indeed status_t is currently defined as an int32, which needs %ld on 32bit systems and %d on 64bit ones. But this avoids the problem alltogether.